runtime/runtime-js/src/main/kotlin/pbandk/protobufjs/Util.kt

```diff
@@ -1,17 +1,34 @@
 package pbandk.protobufjs
 
 import pbandk.asUint8Array
+import pbandk.wkt.Timestamp
+import kotlin.js.Date
+import kotlin.math.floor
 
-val Long.protobufjsLong: dynamic get() {
-    val ret = js("{}")
-    ret.high = this.asDynamic().getHighBits()
-    ret.low = this.asDynamic().getLowBits()
-    return ret
-}
+val Long.protobufjsLong: dynamic
+    get() {
+        val ret = js("{}")
+        ret.high = this.asDynamic().getHighBits()
+        ret.low = this.asDynamic().getLowBits()
+        return ret
+    }
 
 fun Long.Companion.fromProtobufjsLong(l: dynamic) = js("Kotlin").Long.fromBits(l.low, l.high) as Long
 
 object Util {
     fun stringToUtf8(str: String) = ByteArray(util.utf8.length(str)).also { util.utf8.write(str, it.asUint8Array(), 0) }
     fun utf8ToString(bytes: ByteArray) = bytes.asUint8Array().let { util.utf8.read(it, 0, it.length) }
+
+    fun base64ToBytes(str: String) = js("Buffer").from(str, "base64")
+    fun bytesToBase64(bytes: ByteArray) = js("Buffer").from(bytes).toString("base64")
+
+    fun timestampToString(ts: Timestamp.JsonMapper) =
+        Date(((ts.seconds ?: 0) * 1000) + ((ts.nanos ?: 0) / 1_000_000)).toISOString()
```

Comment:
🤔 It's too bad that we lose the nanosecond precision even though both the source (the `Timestamp` object) and the destination (an ISO8601 string) support it, but the intermediate `Date` class does not. There's probably something more involved we could do (rather than using `Date`) to keep the precision, but certainly not worth doing right now.

Reply:
Yeah, I almost did some string hacking, but decided to hold off.